cmd/frontend/graphqlbackend/site_monitoring_test.go

```diff
@@ -0,0 +1,96 @@
+package graphqlbackend
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	prometheus "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+)
+
+func Test_siteMonitoringStatisticsResolver_Alerts(t *testing.T) {
+	mock := NewMockPrometheusQuerier()
+	sampleT := model.Time(time.Now().UTC().Unix())
+	type fields struct {
+		queryValue    model.Value
+		queryWarnings prometheus.Warnings
+		queryErr      error
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		want    []*MonitoringAlert
+		wantErr error
+	}{
+		{"includes alerts with no occurrences", fields{
+			queryValue: model.Matrix{
+				&model.SampleStream{
+					Metric: model.Metric{"name": "hello", "service_name": "world"},
+					Values: []model.SamplePair{{Timestamp: sampleT, Value: model.SampleValue(0)}}},
+			},
+		}, []*MonitoringAlert{{
+			TimestampValue:   DateTime{sampleT.Time().Truncate(time.Hour)},
+			NameValue:        "hello",
+			ServiceNameValue: "world",
+			OccurrencesValue: 0,
+		}}, nil},
```

Comment:
Our current practice is to spell out field names for test structs, which would be much easier to reason when the number of test cases becomes very long, and more readable:
```suggestion
		{
			name: "includes alerts with no occurrences", 
			fields: fields{
			queryValue: model.Matrix{
				&model.SampleStream{
					Metric: model.Metric{"name": "hello", "service_name": "world"},
					Values: []model.SamplePair{{Timestamp: sampleT, Value: model.SampleValue(0)}}},
				},
			}, 
			want: []*MonitoringAlert{{
				TimestampValue:   DateTime{sampleT.Time().Truncate(time.Hour)},
				NameValue:        "hello",
				ServiceNameValue: "world",
				OccurrencesValue: 0,
			}}, 
			wantErr : nil,
		},
```